packages/common/src/utils/memoize.ts

```diff
@@ -0,0 +1,32 @@
+const createCacheKeyFromArgs = (args: any[]) =>
+  args.reduce(
+    (cacheKey, arg) =>
+      (cacheKey += `_${
+        typeof arg === "object" ? JSON.stringify(args) : `${arg}`
+      }_`),
+    ""
+  );
+
+/**
+ * Wrap a function into a memoized version of itself
+ * `const memoizedItemSearch = memoize(searchItems);`
+ * @param fn
+ * @returns
+ */
+export const memoize = <ARGS extends unknown[], RET>(
```

Comment:
❤️

Reply:
yeah - this is a nice function - kinda wondering if we need a means to decache after a specific amount of time though... right now, it's a forever-cache (until you reload the app), but it's good enough for MVP, and can be optimized later